### PR TITLE
GH 3549 mempool allow GCed txs

### LINKED ISSUE #3549: Mempool: allow including already GCed transactions
Garbage collecting transactions is good but the current implementation forbids an already GCed transaction to be added to the mempool. This was initially a spam prevention mechanism which is trivial to work around if a malicious user wants to. Transactions that are already GCed shall be allowed to be included in the mempool as long as they are valid.

### PR DESCRIPTION
**Pull request by velzevur**
Fixes #3549

The work on this PR had been supported by ACF.

### CODE REVIEW

File: apps/aecore/src/aec_tx_gossip_cache.erl [hunk 1]
```diff
@@ -97,14 +97,19 @@ init_state() ->
...
             C1 = maps:put(TxHash, x, maps:remove(Evicted, C)),
             {false, State#state{ evict_q = queue:in(TxHash, Q1), cache = C1, miss = M + 1 }};
         false ->
             {false, State#state{ size = S + 1, evict_q = queue:in(TxHash, Q),
                                  cache = maps:put(TxHash, x, C), miss = M + 1 }}
     end.
+
+max_cache_size() ->
+    aeu_env:user_config_or_env([<<"mempool">>, <<"cache_size">>], aecore,
+                               stratum_enabled, ?CACHE_SIZE).
```
Comment (uwiger) [inline apps/aecore/src/aec_tx_gossip_cache.erl:115]: `stratum_enabled`? This should return a boolean, and doesn't seem to be relevant to the cache size.

Also, shouldn't there be a default value in the schema?

File: apps/aecore/src/aec_tx_pool.erl [hunk 2]
```diff
@@ -168,13 +168,17 @@ push(Tx, Event) ->
 
 -spec push(aetx_sign:signed_tx(), event(), push_timeout()) -> ok | {error, atom()}.
 push(Tx, Event = tx_received, Timeout) ->
+    lager:debug("[domat] tx_received", []),
```
Comment (uwiger) [inline apps/aecore/src/aec_tx_pool.erl:171]: 'domat'?

File: apps/aeutils/priv/aeternity_config_schema.json [hunk 3]
```diff
@@ -53,6 +53,10 @@
                 "nonce_baseline" : {
                     "description" : "Maximum nonce accepted when pubkey is not present in state",
                     "type" : "integer"
+                },
+                "cache_size" : {
+                    "description" : "Maximum number of recently received cached transactions",
+                    "type" : "integer"
```
Comment (uwiger) [inline apps/aeutils/priv/aeternity_config_schema.json:59]: default?

**Review by uwiger — CHANGES_REQUESTED** [review]

File: apps/aecore/src/aec_tx_pool.erl (hunk 2 above)
Comment (velzevur) [reply apps/aecore/src/aec_tx_pool.erl:171]: oh no, my secret logging message is revealed. Again...

File: apps/aeutils/priv/aeternity_config_schema.json (hunk 3 above)
Comment (velzevur) [reply apps/aeutils/priv/aeternity_config_schema.json:59]: Well the rest in this section don't have defaults...

**Review by uwiger — DISMISSED** [review]

**Review by hanssv — APPROVED** [review]

**Review by hanssv — APPROVED** [review]

**Review by cytadela8 — APPROVED** [review]
